website/content/docs/api-gateway/configuration/gatewayclassconfig.mdx

```diff
@@ -0,0 +1,196 @@
+---
+layout: docs
+page_title: Consul API Gateway GatewayClassConfig
+description:  >-
+  Consul API Gateway GatewayClassConfig
+---
+
+# GatewayClassConfig
+
+This topic provides full details about the `GatewayClassConfig` resource
+
+## Introduction
+
+The `GatewayClassConfig` object contains Consul API Gateway-related configuration parameters. Apply the parameters by adding the [`GatewayClass`](#gatewayclass) object to your Kubernetes values file and specifying the name of the `GatewayClassConfig`.
+
+## Configuration model
+
+* [`consul`](#consul): object | optional
+  * [`address`](#consul-address): string | optional
+  * [`authentication`](#consul-authentication): object | optional
+    * [`account`]([#consul-authentication-account): string | optional
+    * [`managed`](#consul-authentication-managed): bool  | optional
+    * [`method`](#consul-authentication-method): string | optional
+    * [`namespace`](#consul-authentication-namespace): string  | optional
+  * [`ports`](#consul-ports): object  | optional
+    * [`grpc`](#consul-ports-grpc): integer | optional
+    * [`http`](#consul-ports-http): integer | optional
+  * [`scheme`](#consul-scheme): string  | optional
+* [`copyAnnotations`](#copyAnnotations): object | optional
+  * [`service`](#copyAnnotations-service): array of strings | optional
+* [`deployment`](#deployment): object | optional
+  * [`defaultInstances`](#deployment-defaultinstances): integer | optional
+  * [`maxInstances`](#deployment-maxinstances): integer | optional
+  * [`minInstances`](#deployment-mininstances): integer | optional
+* [`image`](#image): object | optional
+  * [`consulAPIGateway`](#image-consulapigateway): string | optional
+  * [`envoy`](#image-envoy): string | optional
+* [`logLevel`](#loglevel): string | optional
+* [`nodeSelector`](#nodeselector): string | optional
+* [`serviceType`](#servicetype): string | optional
+* [`useHostPorts`](#usehostports): boolean | optional
+
+## Specification
+
+This topic provides details about the configuration parameters.
+
+### consul
```

Comment:
That's mostly there, but I'd refine it to something like "Specifies configurations that enable Consul API Gateway to interact with Consul" Since Kubernetes Pods is a bit vague.

I'd say it's optional, at least for now. I've never specified any of these values in my tests as the default behavior works out of the box with the helm deployment.